app/code/Magento/Customer/Model/Address/AbstractAddress.php

```diff
@@ -285,6 +285,7 @@ public function setData($key, $value = null)
             foreach ($value as &$attribute) {
                 $attribute = is_array($attribute) ? $attribute : $attribute->__toArray();
                 $attribute = $this->processCustomAttribute($attribute);
+                $attribute = $attribute['value'] ?? null;
```

Comment:
To make the code more consistent, I would strongly recommend making this fix a part of the following method 

https://github.com/rogyar/magento2/blob/91d61bdc5e5b85dcc5b51f7b539055e187a16790/app/code/Magento/Customer/Model/Address/AbstractAddress.php#L684